### DISCUSSION TITLE
Docker. Data are LOST

### DISCUSSION THREAD

**Discussion by StarGuardian** [firefly-iii/firefly-iii#9177] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I installed Firefly using docker compose as it described in documentation.
Run the server using `docker compose up -d` command and everything worked about 3 weeks.
Then I used command `docker compose down` to stop Firefly.
And after I started it again (using `docker compose up`) I found out there is no my data!

### Debug information

Debug information generated at 2024-08-26 13:06:37 for Firefly III version **v6.1.19**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.19 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.9 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td>mysql, <strong>*pgsql*</strong>, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1113, base <span>#</span>86</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Asia/Yekaterinburg + Asia/Yekaterinburg</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-08-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-08-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0</td></tr></tbody></table>

### Expected behaviour

Firefly should keep user's data when container is recreated

### Steps to reproduce

1. Install Firefly using docker
2. Run containers `docker compose up -d`
3. Add some data to Firefly
4. Stop and remove containers (but not volume!) `docker compose down`
5. Run containers again `docker compose up -d`
6. Data are lost!

**Comment by StarGuardian** [comment]
I investigated the issue.
[Installation instruction](https://docs.firefly-iii.org/how-to/firefly-iii/installation/docker/) in section "Change to postgresql" describes following steps:
1. Change .env file
2. Change the docker-compose.yml file to point to postgres instead of mariadb:lts
But essential step: change mountpoint of volume is missed!!!

So I had in my docker-compose.yaml:
```
    volumes:
      - firefly_iii_db:/var/lib/mysql
```
instead of `- firefly_iii_db:/var/lib/postgresql/data`

I suppose it should be added to documentation.

**Reply by JC5** [reply]
Thanks, I just did!
